src/prpy/base/robot.py

```diff
@@ -284,15 +284,33 @@ def ViolatesVelocityLimits(self, traj):
         return False
 
     def _PlanWrapper(self, planning_method, args, kw_args):
-        # Call the planner.
-        traj = planning_method(self, *args, **kw_args)
 
-        # Strip inactive DOFs from the trajectory.
+        # Call the planner.
         config_spec = self.GetActiveConfigurationSpecification()
-        openravepy.planningutils.ConvertTrajectorySpecification(traj, config_spec)
+        result = planning_method(self, *args, **kw_args)
+
+        # Define the post processing steps for the trajectory.
+        def postprocess_trajectory(traj, kw_args):
+
+            # Strip inactive DOFs from the trajectory.
+            openravepy.planningutils.ConvertTrajectorySpecification(
+                traj, config_spec
+            )
+
+            # Optionally execute the trajectory.
+            if 'execute' not in kw_args or kw_args['execute']:
+                return self.ExecuteTrajectory(traj, **kw_args)
+            else:
+                return traj
```

Comment:
I still think we should have the future return the trajectory as it was
executed.  This is immensely useful when doing force moves or move until
touch stuff.

On Thu, Feb 5, 2015 at 11:32 AM, Prasanna Velagapudi <psigen@gmail.com>
wrote:

> Sure, this makes sense for the defer=True case.
>
> On Thu, Feb 5, 2015 at 11:28 AM, Michael Koval <notifications@github.com>
> wrote:
>
>> In src/prpy/base/robot.py
>> <https://github.com/personalrobotics/prpy/pull/42#discussion_r24175418>:
>>
>> > -        openravepy.planningutils.ConvertTrajectorySpecification(traj, config_spec)
>> > +        result = planning_method(self, *args, **kw_args)
>> > +
>> > +        # Define the post processing steps for the trajectory.
>> > +        def postprocess_trajectory(traj, kw_args):
>> > +
>> > +            # Strip inactive DOFs from the trajectory.
>> > +            openravepy.planningutils.ConvertTrajectorySpecification(
>> > +                traj, config_spec
>> > +            )
>> > +
>> > +            # Optionally execute the trajectory.
>> > +            if 'execute' not in kw_args or kw_args['execute']:
>> > +                return self.ExecuteTrajectory(traj, **kw_args)
>> > +            else:
>> > +                return traj
>>
>> On second thought, ExecuteTrajectory should really have two return
>> values (i.e. a tuple):
>>
>>    1. Actual trajectory being executed after any pre-processing (e.g.
>>    cleanup, retiming). This could be a future trajectory if we think the
>>    pre-processing code will be time consuming.
>>    2. Future trajectory status that is evaluated when the trajectory
>>    finishes execution. This includes any error information.
>>
>> All of our existing timeout functionality can be handled by blocking on
>> the future with a timeout.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/personalrobotics/prpy/pull/42/files#r24175418>.
>>
>
>